ballista/core/src/replicator/mod.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use bytes::Bytes;
+use object_store::{path::Path, ObjectStore};
+use tokio::io::AsyncReadExt;
+use tokio::{fs::File, sync::mpsc};
+use tracing::warn;
+
+use crate::error::BallistaError;
+
+pub enum Command {
+    Replicate { path: String },
+}
+
+pub async fn start_replication(
+    base_path: String,
+    object_store: Arc<dyn ObjectStore>,
+    mut receiver: mpsc::Receiver<Command>,
+) -> Result<(), BallistaError> {
+    while let Some(Command::Replicate { path }) = receiver.recv().await {
+        match Path::parse(base_path.as_str()) {
+            Ok(location) => match File::open(path.as_str()).await {
+                Ok(mut file) => {
+                    let size = file.metadata().await.map_or(1024, |m| m.len());
```

Comment:
Currently shuffle files (IPC) do not use compression, which could make them large/inefficient to upload/download

Reply:
We could also change the shuffle writer to use lz4 compression

Reply:
makes sense!